### PR TITLE
Update tcsh to 6.21.00

### PR DESCRIPTION
**Pull request by Mno-hime**
Announcement: https://mailman.astron.com/pipermail/tcsh/2019-May/000018.html

- ~Currently builds only with GCC 8: https://bugs.astron.com/view.php?id=80~ Actually a GCC optimizer bug.
- ~Test fail with `nice`: https://bugs.astron.com/view.php?id=95~

### CODE REVIEW

**Comment by alarcher** [comment]
@Mno-hime Can we merge?

File: components/shell/tcsh/Makefile [hunk 1]
```diff
@@ -67,12 +60,11 @@ COMPONENT_TEST_TRANSFORMS += \
...
-# common targets
 build:		$(BUILD_64)
 
 install:	$(INSTALL_64)
 
 test: 		$(TEST_64)
 
+# Auto-generated dependencies
 REQUIRED_PACKAGES += system/library
-REQUIRED_PACKAGES += text/locale
```
Comment (pyhalov) [inline components/shell/tcsh/Makefile]: Please, check if it is a build dependency

Comment (pyhalov) [reply components/shell/tcsh/Makefile]: @Mno-hime

**Comment by Mno-hime** [comment]
> @Mno-hime Can we merge?

I think we can merge now.

**Pull request merged**